crates/symbolicator/src/services/symbolication.rs

```diff
@@ -218,60 +240,14 @@ impl CfiCacheModules {
             .collect()
     }
 
-    /// Marks a module as scanned for CFI.
-    ///
-    /// If during stack unwinding the module was scanned that means we needed its CFI but we
-    /// did not have it.  We need to keep track of this because it may indicate we didn't
-    /// fetch a CFI file we needed, or we thought a CFI module was missing but it wasn't
-    /// needed.  So this also updates the [`CfiModule::cfi_status`] field.
-    fn mark_scanned(&mut self, code_id: CodeModuleId) {
-        let mut cfi_module = self.inner.entry(code_id).or_insert_with(|| {
-            // We report this error once per missing module in a minidump.
-            sentry::capture_message(
-                "Referenced module not found during initial stack scan",
-                sentry::Level::Error,
-            );
-            CfiModule {
-                cfi_status: ObjectFileStatus::Missing,
-                ..Default::default()
-            }
-        });
-        cfi_module.scanned = true;
-    }
-
-    /// Marks a module's CFI as used during stack unwinding.
-    ///
-    /// This will ensure it does not get the [`ObjectFileStatus::Unused`] status in
-    /// [`CfiCacheModules::finalize`].
-    fn mark_used(&mut self, code_id: CodeModuleId) {
-        let mut cfi_module = self.inner.entry(code_id).or_insert_with(|| {
-            // We report this error once per missing module in a minidump.
-            sentry::capture_message("Module marked as used but not found", sentry::Level::Error);
-            CfiModule {
-                cfi_status: ObjectFileStatus::Missing,
-                ..Default::default()
-            }
-        });
-        cfi_module.cfi_used = true;
-    }
-
-    /// Processes the [`CfiModule::scanned`] information to update the final module status.
-    ///
-    /// All modules which were not scanned should have [`ObjectFileStatus::Unused`].  This
-    /// processes all the [`CfiModule`]s and updates their [`CfiModule::cfi_status`] as
-    /// required.
-    fn finalize(mut self) -> BTreeMap<CodeModuleId, CfiModule> {
-        for cfi_module in self.inner.values_mut() {
-            if !cfi_module.scanned && !cfi_module.cfi_used {
-                cfi_module.cfi_status = ObjectFileStatus::Unused;
-            }
-        }
+    /// Returns the inner Map.
+    fn into_inner(self) -> BTreeMap<CodeModuleId, CfiModule> {
         self.inner
     }
 }
 
 /// A module which was referenced in a minidump and processing information for it.
-#[derive(Debug, Serialize, Deserialize, Default)]
+#[derive(Clone, Debug, Serialize, Deserialize, Default)]
```

Comment:
i assume this is needed because you want to clone the entire BTreeMap that contains this.  This still sounds expensive?